builder/qemu/config.go

```diff
@@ -161,6 +161,66 @@ func (c QemuSMPConfig) getMaxCPUs() int {
 	return totalVCPUs
 }
 
+// Booting in EFI mode
+//
+// Use these options if wanting to boot on a UEFI firmware, as the options to
+// do so are different from what BIOS (default) booting will require.
+type QemuEFIBootConfig struct {
+	// Boot in EFI mode instead of BIOS
+	//
+	// This is required for more modern guest OS
+	//
+	// If either or both of `efi_firmware_code` or `efi_firmware_vars` are
+	// defined, this will implicitely be set to `true`.
+	EnableEFI bool `mapstructure:"efi_boot" required:"false"`
+	// Path to the CODE part of OVMF (or other compatible firmwares)
+	//
+	// The OVMF_CODE.fd file contains the bootstrap code for booting in EFI
+	// mode, and requires a separate VARS.fd file to be able to persist data
+	// between boot cycles.
+	//
+	// If unspecified, and enable_efi is set to true, this will try to load
+	// a plain OVMF_CODE.fd file from `/usr/share/OVMF`
+	OVMFCode string `mapstructure:"efi_firmware_code" required:"false"`
+	// Path to the VARS corresponding to the OVMF code file.
+	//
+	// If unspecified, and enable_efi is set to true, we will try to get the
+	// corresponding VARS.fd file from `/usr/share/`
+	OVMFVars string `mapstructure:"efi_firmware_vars" required:"false"`
+}
+
+func (efiCfg *QemuEFIBootConfig) loadDefaults() {
+	// Auto enable EFI is either of the Code/Vars path is set
+	if efiCfg.OVMFCode != "" || efiCfg.OVMFVars != "" {
+		efiCfg.EnableEFI = true
+	}
+
+	if !efiCfg.EnableEFI {
+		return
+	}
+
+	// If either of Code or Vars is defined, we set the other by replacing
+	// "CODE" or "VARS" by its corollary
+	if efiCfg.OVMFCode == "" && efiCfg.OVMFVars != "" {
+		efiCfg.OVMFCode = strings.Replace(efiCfg.OVMFVars, "VARS", "CODE", -1)
+		return
+	}
+
+	if efiCfg.OVMFVars == "" && efiCfg.OVMFCode != "" {
+		efiCfg.OVMFVars = strings.Replace(efiCfg.OVMFCode, "CODE", "VARS", -1)
+		return
+	}
```

Comment:
Is this always the expected behavior? 

Is it possible that a user would want to load OVMF_VARS.fd from share where outside of `/user/share` but use the default `/usr/share/OVMF_CODE.fd`? 

This logic seems to imply that if vars is set then code must also be set.

Reply:
It is very much possible that a user might want to use a OVMF_CODE.fd and OVMF_VARS.fd from somewhere else than the `/usr/share/` directory yes, the default values try to infer the other from one if either is set simply because of the way the package is structured.
Locally for example, here's what I have:

```
$ ls /usr/share/OVMF/
OVMF_CODE_4M.fd  OVMF_CODE_4M.ms.fd  OVMF_CODE_4M.secboot.fd
OVMF_CODE_4M.snakeoil.fd  OVMF_CODE.fd  OVMF_CODE.ms.fd
OVMF_CODE.secboot.fd  OVMF_VARS_4M.fd  OVMF_VARS_4M.ms.fd
OVMF_VARS_4M.snakeoil.fd  OVMF_VARS.fd  OVMF_VARS.ms.fd
```

So the rationale behind that code is to pick the corresponding VARS/CODE for whichever is set.
But this is probably me overthinking this, we could make the code simpler by setting both when none are set, or print an error message when one is specified, and not the other.
Or default to the `/usr/share/OVMF/OVMF_{CODE,VARS}.fd` when either isn't set.

I'll probably go for the third alternative, and communicate it in the logs so the user is notified when the default value is picked, that'll reduce the risk of surprise.